src_new/util.js

```diff
@@ -1102,6 +1102,40 @@ exports.getAdUnitConfig = function(sizes, currentSlot){
 			return Object.keys(slotConfig['config']).toString().toLowerCase().indexOf(kgpv.toLowerCase()) > -1 ? true : false;
 		}
 	}
+// Returns regex-matched config for kgpv, if not found returns undefined
+	function isAdunitRegex() {
+		var regexKeys = Object.keys(slotConfig['config']);
+		let matchedRegex;
+		regexKeys.forEach((exp) => {
+			try {
+				// Ignores "default" key and RegExp performs case insensitive check
+				if (exp.length > 0 && exp!= CONSTANTS.COMMON.DEFAULT && kgpv.match(new RegExp(exp, "i"))) {
+					matchedRegex = exp;
+					return;
+				}
+			} catch (ex) {
+				refThis.logError(CONSTANTS.MESSAGES.M32 + JSON.stringify(exp));
+			}
+		})
+		if (matchedRegex) {
+			return slotConfig["config"][matchedRegex];
+		} else {
+			return undefined;
+		}
+	}
+	// returns selected MediaConfig
+	function selectSlotConfig(){
+		//exact-match else regex check
+		if(iskgpvpresent()){
+			return slotConfig["config"][kgpv];
+		} else if(isregexEnabled()){
+			return isAdunitRegex();
+		}
+	}
+	// checks if regex is present and enabled
+	function isregexEnabled(){
+		return slotConfig &&(Object.keys(slotConfig).indexOf(CONSTANTS.COMMON.MCONF_REGEX)> -1)&&(slotConfig[CONSTANTS.COMMON.MCONF_REGEX] == true)? true : false;
```

Comment:
Use JS hasOwnProperty instead taking keys and using indexOf to find property is present in object or not

Reply:
Thanks! 
Made respective changes in PR using `refThis.isOwnProperty()` which is existing util function that internally calls `hasOwnProperty()`